src/main/java/productDependency/CreateDependentCategoryPage.java

```diff
@@ -0,0 +1,974 @@
+		CreateDependentCategoryPage sp = new CreateDependentCategoryPage(driver);
+		log.info("click on Dependencies plus icon");
+		Assert.assertTrue(clickDependenciesPlus.isDisplayed());
+		clickDependenciesPlus.click();
+		JavaScriptExec.sleep();
+	}
+
+	private WebElement selectProductCategoryDependency;
+
+	/**
+	 * Method to select product category in dependencies section.
+	 * 
+	 * @throws IOException
+	 */
+	public void selectProductCategoryDependency() throws IOException {
+		CreateDependentCategoryPage sp = new CreateDependentCategoryPage(driver);
+		WebElement PCelement = driver.findElement(By.xpath("//select[@name='product.dependencyItemTypes']"));
+		Select se = new Select(PCelement);
+		se.selectByVisibleText(BasePage.getCellData(xlsxName, sheetName, 3, 0));
+
+	}
+
+	private WebElement selectProductDependency1;
+
+	/**
+	 * Method to select product in dependencies section.
+	 * 
+	 * @throws IOException
+	 */
+	public void selectProductDependency1() throws IOException {
+		CreateDependentCategoryPage sp = new CreateDependentCategoryPage(driver);
+		JavaScriptExec.sleep();
+		driver.findElement(By.xpath("//*[@id='product.dependencyItems']/option[4]")).click();
+		//*[@id="product.dependencyItems"]/option[3]
+		
+		
+		
+
+	}
+
+	private WebElement selectProductPeriod;
+
+	/**
+	 * Method to select product period in dependencies section.
+	 * 
+	 * @throws IOException
+	 */
+	public void selectProductPeriod() throws IOException {
+		CreateDependentCategoryPage sp = new CreateDependentCategoryPage(driver);
+		WebElement Productelement = driver.findElement(By.xpath("//select[@name='period']"));
+		Select se = new Select(Productelement);
+		se.selectByVisibleText(BasePage.getCellData(xlsxName, sheetName, 7, 0));
+
+	}
+	
+	@FindBy(how = How.XPATH, using = "//input[@id='product.dependencyMin']")
+	
+	//*[@id="product.dependencyMin"]
+	private WebElement enterMin;
+
+	/**
+	 * Method to Enter Price Rate.
+	 * 
+	 * @throws IOException
+	 */
+	public void enterMin() throws IOException {
+		CreateDependentCategoryPage sp = new CreateDependentCategoryPage(driver);
+		log.info("Enter Min value");
+		JavaScriptExec.sleep();
+		WaitClass.WaitForElementisDisplay(driver, 10, enterMin);
+		Assert.assertTrue(enterPriceRate.isDisplayed());
+		//enterPriceRate1.clear();
+		enterMin.clear();
+		enterMin.sendKeys("1");
+
+}
+	@FindBy(how = How.XPATH, using = "//*[@id='dependency']/div[2]/div/table/tbody/tr/td[6]/a/img")
+	private WebElement clickDependencyPlusIcon;
+
+	/**
+	 * Method to click on dependency plus icon to add the product.
+	 * 
+	 * @throws IOException
+	 */
+	public void clickDependencyPlusIcon() throws IOException {
+		CreateDependentCategoryPage sp = new CreateDependentCategoryPage(driver);
+		log.info("click on dependency plus icon to add the product.");
+		Assert.assertTrue(clickDependencyPlusIcon.isDisplayed());
+		clickDependencyPlusIcon.click();
+		JavaScriptExec.scrollToElementOnPage(driver, clickDependencyPlusIcon);
```

Comment:
@mayankchhajer In the Selenium we have to find the WebElement 1stly and then we will click on that element, here you are 1stly clicking on the element and then search for the Element. Please correct this.

Reply:
Done